### PR TITLE
ENH: Wrap TransformMeshFilter for BSplineTransform type

### PR DESCRIPTION
**Pull request by tbirdso**
## PR Checklist
- [ ] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [ ] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)
- [x] Added test (or behavior not changed)
- [ ] Updated API documentation (or API not changed)
- [ ] Added [license](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Utilities/KWStyle/ITKHeader.h) to new files (if any)
- [x] Added Python wrapping to new files (if any) as described in [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) Section 9.5
- [ ] Added [ITK examples](https://github.com/InsightSoftwareConsortium/ITKExamples) for all new major features (if any)

Refer to the [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) for
further development details if necessary.

### CODE REVIEW

**Review by dzenanz — COMMENTED** [review]
LGTM

File: Modules/Core/Mesh/wrapping/itkTransformMeshFilter.wrap [hunk 1]
```diff
@@ -14,7 +14,7 @@ itk_wrap_class("itk::TransformMeshFilter" POINTER)
             "itk::Mesh< ${ITKT_${t}},${d} >, itk::Mesh< ${ITKT_${t}},${d} >, itk::Transform< ${ITKT_${tt}}, ${d}, ${d} >")
       endforeach()
     endforeach()
+    itk_wrap_template("M${ITKM_F}${d}M${ITKM_F}${d}BST${ITKM_D}${d}3"
+          "itk::Mesh< ${ITKT_F},${d} >, itk::Mesh< ${ITKT_F},${d} >, itk::BSplineTransform<${ITKT_D},${d},3>")
```
Comment (dzenanz) [inline Modules/Core/Mesh/wrapping/itkTransformMeshFilter.wrap:18]:
```suggestion
          "itk::Mesh< ${ITKT_F},${d} >, itk::Mesh< ${ITKT_F},${d} >, itk::BSplineTransform<${ITKT_D},${d},${d}>")
```

Comment (dzenanz) [reply Modules/Core/Mesh/wrapping/itkTransformMeshFilter.wrap:18]: The fist dimension of the transform should match dimension of the input, the second one that of the output mesh. Here we use `${d}` for both.

**Review by dzenanz — CHANGES_REQUESTED** [review]

File: Modules/Core/Mesh/wrapping/itkTransformMeshFilter.wrap (hunk 1 above)
Comment (tbirdso) [reply Modules/Core/Mesh/wrapping/itkTransformMeshFilter.wrap:18]: The [B-Spline documentation](https://itk.org/Doxygen/html/classitk_1_1BSplineTransform.html) indicates that only the second of the three template arguments describes mesh dimensions, while the third argument is used to determine mesh size: 
> class itk::BSplineTransform< TParametersValueType, NDimensions, VSplineOrder >
...
> This class encapsulates a deformable transform of points from one N-dimensional space to another N-dimensional space. 
...
> mesh size = number of control points - spline order

In that case shouldn't the third argument be fixed separate from mesh dimensions?

This is reflected in the current wrapping of `BSplineTransform`:
> \>itk.BSplineTransform.GetTypes()
\<itkTemplate itk::BSplineTransform>
Options:
  [<itkCType double>, 2, 3]
  [<itkCType double>, 3, 3]

Comment (dzenanz) [reply Modules/Core/Mesh/wrapping/itkTransformMeshFilter.wrap:18]: I didn't even look at the docs, sorry about noise.

**Review by dzenanz — APPROVED** [review]

File: Modules/Core/Mesh/wrapping/itkTransformMeshFilter.wrap (hunk 1 above)
Comment (tbirdso) [reply Modules/Core/Mesh/wrapping/itkTransformMeshFilter.wrap]: No problem, thanks for checking!

File: Modules/Core/Mesh/wrapping/itkTransformMeshFilter.wrap [hunk 2]
```diff
@@ -10,6 +11,8 @@ itk_wrap_class("itk::TransformMeshFilter" POINTER)
       foreach(tt ${transform_type})
         itk_wrap_template("M${ITKM_${t}}${d}M${ITKM_${t}}${d}T${ITKM_${tt}}${d}${d}"
             "itk::Mesh< ${ITKT_${t}},${d} >, itk::Mesh< ${ITKT_${t}},${d} >, itk::Transform< ${ITKT_${tt}}, ${d}, ${d} >")
+        itk_wrap_template("M${ITKM_${t}}${d}M${ITKM_${t}}${d}BST${ITKM_${tt}}${d}3"
+              "itk::Mesh< ${ITKT_${t}},${d} >, itk::Mesh< ${ITKT_${t}},${d} >, itk::BSplineTransform<${ITKT_${tt}},${d},3>")
```
Comment (thewtex) [inline Modules/Core/Mesh/wrapping/itkTransformMeshFilter.wrap:15]: We should not need to wrap the derived class, also.

**Review by thewtex — CHANGES_REQUESTED** [review]
We should not need

**Comment by tbirdso** [comment]
Closed after discussion of redundancy. To paraphrase @thewtex , the correct approach to using `TransformMeshFilter` with descendants of `itk::Transform` in Python is to rely on the functional interface `output_mesh = itk.transform_mesh_filter(input_mesh, transform=my_transform)`. No additional wrapping is required.

**Pull request closed without merging**